### PR TITLE
ISSD-1489 [Marketplace] Updated Complete flow of clarity solutions to use Steps on page

### PR DESCRIPTION
**Pull request by welitonpe**
**[ISSD-971](https://liferay.atlassian.net/jira/software/c/projects/ISSD/boards/789?selectedIssue=ISSD-971)**

### CODE REVIEW

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by welitonpe** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useAccountForm.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			formState: {errors, isValid},
+			getValues,
+			handleSubmit,
+			hasAllValidations,
+			register,
+			setValue,
+			singleAccount,
+			watch,
+		},
+	};
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useAccountForm.ts:107]:
```suggestion
	return form
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useAccountForm.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			setValue('accounts', userAccounts);
+			setValue('accountQuantity', userAccounts.length);
+		})();
+	}, [fetchAccount, myUserAccount, setStep, setValue, step]);
+
+	const form = watch();
+
+	const agreeToTermsAndConditions = form['agreeToTermsAndConditions'];
+
+	const hasAllValidations = agreeToTermsAndConditions && isValid;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useAccountForm.ts:93]: 1. Not watch the entire form

```suggestion
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	type?: string;
+};
+
+const accountTypes = {
+	BUSINESS: 'business',
+	PERSON: 'person',
+};
+
+type HandleAccountProps = {
+	mutateMyUserAccout: KeyedMutator<any>;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts:29]:
```suggestion
	mutateMyUserAccount: KeyedMutator<any>;
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	myUserAccount: UserAccount;
+};
+
+const useHandleAccount = ({
+	mutateMyUserAccout,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts:34]:
```suggestion
	mutateMyUserAccount,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+	myUserAccount,
+}: HandleAccountProps) => {
+	const addUserAccountInAccount = async (data: Account) => {
+		if (myUserAccount && data?.externalReferenceCode) {
+			await fetcher.post(
+				`o/headless-admin-user/v1.0/accounts/by-external-reference-code/${data?.externalReferenceCode}/user-accounts/by-external-reference-code/${myUserAccount.externalReferenceCode}`
+			);
+
+			mutateMyUserAccout(
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts:43]:
```suggestion
			mutateMyUserAccount(
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+			.patch(`o/headless-admin-user/v1.0/accounts/${accountId}`, data)
+			.catch((error) => console.error(error));
+
+		await addUserAccountInAccount(response);
+	};
+
+	const createAccount = async (data: FormType) => {
+		const response: Account = await fetcher
+			.post('o/headless-admin-user/v1.0/accounts', data)
+			.catch((error) => console.error(error));
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts:69]:
```suggestion
		const response = await fetcher<Account>
			.post('o/headless-admin-user/v1.0/accounts', data)
			.catch((error) => console.error(error));
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+		await addUserAccountInAccount(response);
+
+		return response;
+	};
+
+	const formDataTransform = (form: UserForm) => {
+		const submitForm: FormType = {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedSolutions/hooks/useHandleAccount.ts:77]:
```suggestion
		const submitForm = {
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/MarketplaceContext.tsx [hunk 8]
```diff
@@ -4,19 +4,21 @@
...
 
 type ContextType = {
 	channel: Channel;
+	mutateMyUserAccout: KeyedMutator<UserAccount | undefined>;
 	myUserAccount: UserAccount;
 };
 
 const MarketplaceContext = createContext<ContextType>({
 	channel: {} as Channel,
+	mutateMyUserAccout: {} as any,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/MarketplaceContext.tsx:21]:
```suggestion
	mutateMyUserAccount: () => null as KeyedMutator<UserAccount | undefined>,
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/app-banner/index.html [hunk 9]
```diff
@@ -1,11 +1,35 @@
...
+[#assign productId = configuration.productId /]
+
+[#if productId?has_content]
+	[#assign newProductId = productId?number/]
+[/#if]
+
+[#if channelId?has_content && productId?has_content && newProductId?has_content]
+	[#assign
+	product = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels/${channelId}/products/${newProductId+1}?accountId=-1&nestedFields=productSpecifications,categories")
+	productCategories = product.categories![] /]
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/app-banner/index.html:24]:
```suggestion
	[#assign
        product = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels/${channelId}/products/${newProductId+1}?accountId=-1&nestedFields=productSpecifications,categories")
        productCategories = product.categories![] 
    /]
```

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/app-banner/index.html [hunk 10]
```diff
@@ -1,11 +1,35 @@
-[#assign publisherEmail = configuration.publisherEmail]
+[#assign publisherEmail = configuration.publisherEmail /]
+	[#if themeDisplay?has_content]
+	[#assign scopeGroupId = themeDisplay.getScopeGroupId() /]
+[/#if]
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/app-banner/index.html:4]:
```suggestion
[#assign publisherEmail = configuration.publisherEmail /]
[#if themeDisplay?has_content]
	[#assign scopeGroupId = themeDisplay.getScopeGroupId() /]
[/#if]
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by kevenleone** [comment]
resent: https://github.com/brianchandotcom/liferay-portal/pull/143103

**Pull request closed without merging**